flake.nix

```diff
@@ -0,0 +1,39 @@
+
+{
+  description = "A basic Rust devshell";
```

Comment:
We should probably update this to reference spin, but that can be done as a follow up

Reply:
I mentioned spin in the new commit